internal/ask/form/form.go

```diff
@@ -102,6 +102,8 @@ func Upsert(context interface{}, db *db.DB, form *Form) error {
 	if form.ID.Hex() == "" {
 		form.ID = bson.NewObjectId()
 		isNewForm = true
+		form.DateCreated = time.Now()
+		form.DateUpdated = time.Now()
 	}
```

Comment:
Just move this down a line, and leave UpdateStats untouched. It still performs an update there, but for changing form contents we are also updating the form, hence we should also update the date updated.